src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -1577,22 +1578,135 @@ AcquireExecutorShardLocksForExecution(DistributedExecution *execution)
 		return;
 	}
 
+	bool parallelExecutionNotPossible =
+		list_length(taskList) == 1 || ShouldRunTasksSequentially(taskList);
+
+	bool anyAnchorTableIsReplicated = AnyAnchorTableIsReplicated(taskList);
```

Comment:
anchor table is an unfamiliar concept to me

I feel like AnyAnchorTableIsReplicated is trying to do 2 separate things (check for reference tables, check for replicated tables), could we separate those out?